graphs/eacs/src/test/kotlin/com/anaplan/engineering/azuki/graphs/eacs/BEH7.kt

```diff
@@ -0,0 +1,136 @@
+package com.anaplan.engineering.azuki.graphs.eacs
+
+import com.anaplan.engineering.azuki.core.runner.Eac
+import com.anaplan.engineering.azuki.core.system.BEH
+import com.anaplan.engineering.azuki.graphs.adapter.api.GraphBehaviours
+import com.anaplan.engineering.azuki.graphs.adapter.api.GraphFunctions
+import com.anaplan.engineering.azuki.graphs.dsl.GraphScenario
+import com.anaplan.engineering.azuki.graphs.graphA
+
+@BEH(GraphBehaviours.GetSimpleCycleCount, GraphFunctions.GetSimpleCycleCount, """
+    Determine the number of cycles in a given directed graph
+""")
+class BEH7 : GraphScenario() {
+
+    @Eac("A cycle is a continuous path which starts and ends at the same vertex")
+    fun hasOneCycleDirected() {
+        given {
+            thereIsADirectedGraph(graphA) {
+                edge("a", "b")
+                edge("b", "c")
+                edge("d", "a")
+                edge("e", "d")
+                edge("c", "e")
+            }
+        }
+        then {
+            getSimpleCycleCount(graphA, 1)
+        }
+    }
+
+    @Eac("When there is no continuous path which starts and ends at the same vertex there are no cycles")
+    fun hasNoCyclesDirected() {
+        given {
+            thereIsADirectedGraph(graphA) {
+                edge("a", "b")
+                edge("b", "c")
+                edge("d", "a")
+                edge("e", "d")
+            }
+        }
+        then {
+            getSimpleCycleCount(graphA, 0)
+        }
+    }
+
+    @Eac("""
+        When the given graph is directed, any path which would be considered a cycle in an undirected graph but
+        where two edges start (or end) at the same vertex is not counted as a cycle
+        """)
+    fun hasReversedEdgeDirected() {
+        given {
+            thereIsADirectedGraph(graphA) {
+                edge("a", "b")
+                edge("b", "c")
+                edge("d", "a")
+                edge("e", "d")
+                edge("e", "c")
+            }
+        }
+        then {
+            getSimpleCycleCount(graphA, 0)
+        }
+    }
+
+    @Eac("When the given graph is empty, return 0")
+    fun hasEmptyDirectedGraph() {
+        given {
+            thereIsADirectedGraph(graphA) {}
+        }
+        then {
+            getSimpleCycleCount(graphA, 0)
+        }
+    }
+
+    @Eac("When the given graph has multiple individual cycles that share edges, each cycle is counted")
```

Comment:
🚫 I don't like this phrasing because taken at face value it is implies that having an edge in common is sufficient for two cycles to be counted separately. I would prefer something like:

> Having one or more edges in common does not prevent two simple cycles from being counted separately

If you don't like the passive voice, you can phrase it along the lines of:

> Two simple cycles can still be counted separately if they share one or more edges